### PR TITLE
Nonsilence operator

### PR DESCRIPTION
**Pull request by szalpal**
This operator is required for RNN-T support in Dali.
Operator performs silence detection in an audio buffer. It returns begin index of nonsilent region and its length.

#### Why we need this PR?
*Pick one, remove the rest*
- To support RNN-T in Dali

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     *Added new operator*
 - Validation and testing:
     *Python and GTest unit tests*
 - Documentation (including examples):
     *Documentation in tests*

**JIRA TASK**: *NA*

### CODE REVIEW

**Comment by szalpal** [comment]
!build

File: dali/operators/audio/nonsilence_op.h [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ * @return (begin index, length)
+ */
+template<typename T>
+std::pair<int, int> DetectNonsilenceRegion(span<const T> buffer, T cutoff) {
+  int begin = -1;
+  int end = buffer.size();
+  while (begin < end && buffer[++begin] < cutoff);  // NOLINT
+  if (begin == end) return {-1, 0};
+  while (buffer[--end] < cutoff);  // NOLINT
+  return {begin, end - begin + 1};
```
Comment (klecki) [inline dali/operators/audio/nonsilence_op.h]: Why don't you use the same implementation as in the docs.
It's much easier to read than those tricky one-liner whiles.
The lint complained here, probably for valid reason.

File: dali/operators/audio/nonsilence_op.cc [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+            break
+    length = end - begin + 1
+    return begin, length
+```
+`Input`: 1-D audio buffer
+`Output[0]`: Begin index of nonsilent region
+`Output[1] >= 0`: Length of nonsilent region<br>
+If `Output[1] == 0`, `Output[0]` value is undefined
+)code")
+                .NumInput(1)
```
Comment (klecki) [inline dali/operators/audio/nonsilence_op.cc]: You can use docstring like Reshape to indicate the dimensionality of the input.

File: dali/operators/audio/nonsilence_op.cc [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                .NumOutput(detail::kNumOutputs)
+                .AddArg(detail::kCutoff, R"code(Everything below this value will be regarded as silence)code", DALI_FLOAT);
+
+DALI_REGISTER_OPERATOR(NonsilenceRegion, NonsilenceOperatorCpu, CPU);
+
+#define NONSILENCE_TYPES (uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, uint64_t, int64_t, float, double)  // NOLINT
+
```
Comment (klecki) [inline dali/operators/audio/nonsilence_op.cc]: Any check for 1D in the input?
What about channels?

File: dali/operators/audio/nonsilence_op.h [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+static_assert(std::is_integral<OutputType>::value,
+              "Operator return indices, thus OutputType shall be integral");
+
+/**
+ * Detects nonsilence region in provided 1-D audio buffer
+ * @param cutoff Everything below this value will be reagrded as silence
+ * @return (begin index, length)
+ */
+template<typename T>
+std::pair<int, int> DetectNonsilenceRegion(span<const T> buffer, T cutoff) {
```
Comment (klecki) [inline dali/operators/audio/nonsilence_op.h]: Some named type maybe? :P

File: dali/operators/audio/nonsilence_op.h [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    return true;
+  }
+
+
+  USE_OPERATOR_MEMBERS();
+  const float cutoff_;
+};
+
+
+class NonsilenceOperatorCpu : public NonsilenceOperator<CPUBackend> {
```
Comment (klecki) [inline dali/operators/audio/nonsilence_op.h]: Why those two levels of inheritance?

**Pull request closed without merging**